src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -270,14 +281,17 @@ ctTypeMeta f (CTOut meta) = CTOut <$> f meta
 -- It includes meta data which are not part of Cardano, too
 -- (Flow type: transactionType)
 data CTx = CTx
-    { ctId            :: CTxId
-    , ctAmount        :: CCoin
-    , ctConfirmations :: Word
-    , ctType          :: CTType -- it includes all "meta data"
+    { ctId            :: !CTxId
+    , ctAmount        :: !CCoin -- contains what you'd expect in reasonable cases
+    -- `Maybe` is an implementation detail here
+    , ctFrom          :: !(Maybe [(CAddress, CCoin)])
```

Comment:
hm, why maybe? :/ 

Frontend depends on this implementation detail now tho. I guess transaction should always have a sender, so I don't see a reason for `Maybe` here?